include/ccf/historical_queries_adapter.h

```diff
@@ -83,9 +83,148 @@ namespace ccf::historical
+    const auto committed_view = consensus->get_view(committed_seqno);
+
+    const auto tx_status = ccf::evaluate_tx_status(
+      view, seqno, tx_view, committed_view, committed_seqno);
+    switch (tx_status)
+    {
+      case ccf::TxStatus::Unknown:
+      case ccf::TxStatus::Pending:
+        error_reason = fmt::format(
+          "Only committed transactions can be queried. Transaction {}.{} "
+          "is "
+          "{}",
+          view,
+          seqno,
+          ccf::tx_status_to_str(tx_status));
+        return HistoricalTxStatus::PendingOrUnknown;
+      case ccf::TxStatus::Invalid:
+        error_reason = fmt::format(
+          "Only committed transactions can be queried. Transaction {}.{} "
+          "is "
+          "{}",
+          view,
+          seqno,
+          ccf::tx_status_to_str(tx_status));
+        return HistoricalTxStatus::Invalid;
+      case ccf::TxStatus::Committed:;
+    }
+
+    return HistoricalTxStatus::Valid;
+  };
+
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wunused-function"
 
+  static ccf::endpoints::EndpointFunction adapter_v2(
+    const HandleHistoricalQuery& f,
+    AbstractStateCache& state_cache,
+    const CheckHistoricalTxStatus& available,
+    const TxIDExtractor& extractor = txid_from_header)
+  {
+    return [f, &state_cache, available, extractor](
+             endpoints::EndpointContext& args) {
+      // Extract the requested transaction ID
+      ccf::TxID target_tx_id;
+      {
+        const auto tx_id_opt = extractor(args);
+        if (tx_id_opt.has_value())
+        {
+          target_tx_id = tx_id_opt.value();
+        }
+        else
+        {
+          return;
+        }
+      }
+
+      // Check that the requested transaction ID is available
+      {
+        auto error_reason = fmt::format(
+          "Transaction {} is not available.", target_tx_id.to_str());
+        auto is_available =
+          available(target_tx_id.view, target_tx_id.seqno, error_reason);
+        switch (is_available)
+        {
+          case HistoricalTxStatus::Error:
+            args.rpc_ctx->set_error(
+              HTTP_STATUS_INTERNAL_SERVER_ERROR,
+              ccf::errors::TransactionPendingOrUnknown,
+              std::move(error_reason));
+            return;
+          case HistoricalTxStatus::PendingOrUnknown:
+            // Set header No-Cache
+            args.rpc_ctx->set_response_header(
+              http::headers::CACHE_CONTROL, "no-cache");
+            args.rpc_ctx->set_error(
+              HTTP_STATUS_NOT_FOUND,
+              ccf::errors::TransactionPendingOrUnknown,
+              std::move(error_reason));
+            return;
+          case HistoricalTxStatus::Invalid:
+            args.rpc_ctx->set_error(
+              HTTP_STATUS_NOT_FOUND,
+              ccf::errors::TransactionInvalid,
+              std::move(error_reason));
+            return;
+          case HistoricalTxStatus::Valid:;
```

Comment:
```suggestion
          case HistoricalTxStatus::Error:
          {
            args.rpc_ctx->set_error(
              HTTP_STATUS_INTERNAL_SERVER_ERROR,
              ccf::errors::TransactionPendingOrUnknown,
              std::move(error_reason));
            return;
          }
          case HistoricalTxStatus::PendingOrUnknown:
          {
            // Set header No-Cache
            args.rpc_ctx->set_response_header(
              http::headers::CACHE_CONTROL, "no-cache");
            args.rpc_ctx->set_error(
              HTTP_STATUS_NOT_FOUND,
              ccf::errors::TransactionPendingOrUnknown,
              std::move(error_reason));
            return;
          }
          case HistoricalTxStatus::Invalid:
          {
            args.rpc_ctx->set_error(
              HTTP_STATUS_NOT_FOUND,
              ccf::errors::TransactionInvalid,
              std::move(error_reason));
            return;
          }
          case HistoricalTxStatus::Valid:
          {
            continue;
          }
```

Reply:
I mean, ok I guess, but that `continue`...